interface/reports/collections_report.php

```diff
@@ -762,11 +764,7 @@ function checkAll(checked) {
         if (! $duncount) {
             for ($i = 1; $i <= 3; ++$i) {
                 $tmp = SLEOB::arGetPayerID($patient_id, $svcdate, $i);
-
                 if (empty($tmp)) {
-                    break;
```

Comment:
hi @juggernautsei , nice pull request, not sure why you didn't close up the if statement here, but will put it back in to continue review else can't load report page

Reply:
Found it and fixed. Typo!

Reply:
This seems odd that we are only setting $payerids[] to $tmp when it is empty?? I don't know this code well, and just want to make sure that is what you mean to be doing here.

Reply:
This code predates me. It has been that way for years. That is why I am raising questions about the logic here to see if there is a better way to do this. The other thing that is odd to me is counting statements. A practice would have to know to send a statement every time a payment came in for this code to work correctly.  The due patient and the due insurance should not be connected with sending a statement. The practice that I am working with is waiting until all insurances have paid to before sending a statement. So, I asked them to give me the pseudo logic to determine when a balance is still due by insurance and when should it drop to the patient responsibility.

Reply:
> A practice would have to know to send a statement every time a payment came in for this code to work correctly. 

since one generally runs the bills on a schedule, say once a month, and the fees->posting->pt due will generate a list of those who should receive bills, this will be satisfied

think Brady's right to put the `break` back into that if statement